02-analytics/technical.md

```diff
@@ -0,0 +1,377 @@
+|---------------------|----------------|-----------------------------------------------|
+| OS                  | os             | Automatically managed by SDKs                 |
+| OS Language         | locale         | Automatically managed by SDKs                 |
+| Network Info        | networkInfo    | Automatically managed by SDKs                 |
+| Device Info         | deviceInfo     | Automatically managed by SDKs                 |
+| Anonymous ID        | anonymousId    | Automatically managed by SDKs                 |
+| Channel             | channel        | Automatically managed by SDKs                 |
+| Timezone            | timezone       | Automatically managed by SDKs                 |
+| Screen data         | screen         | Automatically managed by SDKs                 |
+| App Language        | appLanguage    |                                               |
+| Release Version     | releaseVersion | App version, or build number/git hash for web |
+| Person analytics ID | presonId       | For logged in users, see below                |
+
+### Person analytics ID
+
+You can get this ID by sending a token to TODO. The returned value is an hash
+of the userID + a secret. This ensures that we are able to link all data to one
+user from different systems, while preventing anyone from being able to reverse
+the process or generate analytics IDs if they do not posses the secret.
+
+## Identify
+
+*When*: On login + When you request a new refresh token from Auth0
+*Reason*: Being able to connect anonymous events to a user
+
+### API
+
+Use the `/identify` endpoint. Docs:			https://docs.rudderstack.com/rudderstack-api/api-specification/rudderstack-spec/identify
+
+### Data
+
+This call accepts much more data, but we do not want to send it here. The additional
+data will be automatically injected by RudderStack, and delivered to the needed targets.
+
+| Data         | Name        | Comments                                                                         |
+|--------------|-------------|----------------------------------------------------------------------------------|
+| Anonymous ID | anonymousId | Automatically managed by SDKs                                                    |
+| PersonID     | id          | This is the actual person ID. It will be used for data lookup and then discarded |
+
+## Screen/Page View
+
+*When*: On every page/screen the user opens. If relevant also on popups if they
+can be considered a separate screen.
+
+*Reason*: It will allow us to see how used the various pages are and estimate how
+much time users spend on various screens.
+
+### API
+
+Use the `/page` endpoint on web. Docs: https://docs.rudderstack.com/rudderstack-api/api-specification/rudderstack-spec/page
+Use the `/screen` endpoint in app. Docs: https://docs.rudderstack.com/rudderstack-api/api-specification/rudderstack-spec/screen
+
+### Data
+
+The rest of the relevant additional data is automatically collected by the SDK.
```

Comment:
Whenever you refer to SDK that does something, it is not relevant for mobile app, because there is no SDK. We have to write it on our own. For this reason please clarify in details SDK functionality.

Reply:
See notes on Teams about SDK.

Reply:
Thank you, I replied on teams.

Reply:
For the record: Suggested was the use of Segment xamarin plugin. Turns out it's outdated and should not be used anymore. At the same time it does not provide any additional data.

Reply:
@michalbcc can you check the latest version if that contains ok information?